src/swupd_lib/target_root.c

```diff
@@ -632,6 +665,25 @@ static int rename_all_files_to_final(struct list *updates)
 		progress_report(list_length + complete, list_length * 2);
 	}
 
+	dirs_list = list_head(dirs_to_remove);
+	while (dirs_list) {
+		struct file *file;
+		file = dirs_list->data;
+		dirs_list = dirs_list->next;
+
+		ret = rename_staged_file_to_final(file);
+		if (ret != 0) {
+			ret = move_to_timestamp(file->filename);
+			if (ret != 0) {
+				error("Could not move file to backup location: %s\n", file->filename);
```

Comment:
No, this isn't "fatal" it would happen when a directory swupd thinks should be deleted but user content in it can't be renamed for some reason likely disk related I'd expect. The directory would be left with whatever user content was in it (that's what the filename would be in this case). Updates after will still work fine too but swupd repair --picky will try and delete it.